libewf/libewf_single_files.c

```diff
@@ -988,50 +1533,2314 @@ int libewf_single_files_parse_record_values(
+						error,
+						LIBCERROR_ERROR_DOMAIN_MEMORY,
+						LIBCERROR_MEMORY_ERROR_INSUFFICIENT,
+						"%s: unable to create ip_address.",
+						function);
+
+					goto on_error;
+				}
+				if (narrow_string_copy(
+					single_file_source->ip_address,
+					value_string,
+					value_string_size - 1) == NULL)
+				{
+					libcerror_error_set(
+						error,
+						LIBCERROR_ERROR_DOMAIN_MEMORY,
+						LIBCERROR_MEMORY_ERROR_SET_FAILED,
+						"%s: unable to set ip_address.",
+						function);
+
+					goto on_error;
+				}
+				single_file_source->ip_address[value_string_size - 1] = 0;
+
+				single_file_source->ip_address_size = value_string_size;
+			}
+			else if ((type_string[0] == (uint8_t) 's')
+				  && (type_string[1] == (uint8_t) 'i'))
+			{ // Static IP (bool)?
+				/* 1 if static IP
+				 * empty otherwise
+				 */
+				 if (value_string == NULL)
+				 {
+					 single_file_source->static_ip = 0;
+				 }
+				 else if ((value_string_size == 2)
+					 && (value_string[0] == (uint8_t) '1'))
+				 {
+					 single_file_source->static_ip = 1;
+				 }
+
+			}
+			else if ((type_string[0] == (uint8_t) 'm')
+				  && (type_string[1] == (uint8_t) 'a'))
+			{
+				single_file_source->mac_address = (uint8_t *)memory_allocate(
+					sizeof(uint8_t) * value_string_size);
+
+				if (single_file_source->mac_address == NULL)
+				{
+					libcerror_error_set(
+						error,
+						LIBCERROR_ERROR_DOMAIN_MEMORY,
+						LIBCERROR_MEMORY_ERROR_INSUFFICIENT,
+						"%s: unable to create mac_address.",
+						function);
+
+					goto on_error;
+				}
+				if (narrow_string_copy(
+					single_file_source->mac_address,
+					value_string,
+					value_string_size - 1) == NULL)
+				{
+					libcerror_error_set(
+						error,
+						LIBCERROR_ERROR_DOMAIN_MEMORY,
+						LIBCERROR_MEMORY_ERROR_SET_FAILED,
+						"%s: unable to set mac_address.",
+						function);
+
+					goto on_error;
+				}
+				single_file_source->mac_address[value_string_size - 1] = 0;
+
+				single_file_source->mac_address_size = value_string_size;
+			}
+		}
+		else if (type_string_size == 2)
+		{
+			if (type_string[0] == (uint8_t) 'n')
+			{
+				single_file_source->name = (uint8_t *)memory_allocate(
+					sizeof(uint8_t) * value_string_size);
+
+				if (single_file_source->name == NULL)
+				{
+					libcerror_error_set(
+						error,
+						LIBCERROR_ERROR_DOMAIN_MEMORY,
+						LIBCERROR_MEMORY_ERROR_INSUFFICIENT,
+						"%s: unable to create name.",
+						function);
+
+					goto on_error;
+				}
+				if (narrow_string_copy(
+					single_file_source->name,
+					value_string,
+					value_string_size - 1) == NULL)
+				{
+					libcerror_error_set(
+						error,
+						LIBCERROR_ERROR_DOMAIN_MEMORY,
+						LIBCERROR_MEMORY_ERROR_SET_FAILED,
+						"%s: unable to set name.",
+						function);
+
+					goto on_error;
+				}
+				single_file_source->name[value_string_size - 1] = 0;
+
+				single_file_source->name_size = value_string_size;
+			}
+			else if (type_string[0] == (uint8_t) 'p')
```

Comment:
@jtsylve this does not make sense in the context of source, was this just copied from file entry?

Reply:
I disagree with how `'p'` is interpreted here, I've updated the documentation.